Source/Services/Beam/beam_manager_impl.cpp

```diff
@@ -247,6 +275,88 @@ beam_manager_impl::init_worker(
     return;
 }
 
+
+void xbox::services::beam::beam_manager_impl::process_messages_worker()
+{
+    static const int chunkSize = 10;
+    while (m_processing)
+    {
+        {
+            std::lock_guard<std::recursive_mutex> lock(m_messagesLock);
+            for (int i = 0; i < chunkSize && !m_unhandledFromService.empty(); i++)
+            {
+                std::shared_ptr<beam_rpc_message> currentMessage = m_unhandledFromService.front();
+                m_unhandledFromService.pop();
+                try
+                {
+                    if (currentMessage->m_json.has_field(RPC_TYPE))
+                    {
+                        string_t messageType = currentMessage->m_json[RPC_TYPE].as_string();
+
+                        if (0 == messageType.compare(RPC_REPLY))
+                        {
+                            process_reply(currentMessage->m_json);
+                        }
+                        else if (0 == messageType.compare(RPC_METHOD))
+                        {
+                            process_method(currentMessage->m_json);
+                        }
+                        else
+                        {
+                            LOGS_INFO << "Unexpected message from service";
+                        }
+                    }
+                }
+                catch (std::exception e)
+                {
+                    LOGS_ERROR << "Failed to parse incoming socket message";
+                }
+            }
+        }
+
+        if (m_webSocketConnection && m_webSocketConnection->state() == beam_web_socket_connection_state::connected)
+        {
+            std::lock_guard<std::recursive_mutex> lock(m_messagesLock);
+            for (int i = 0; i < chunkSize && !m_pendingSend.empty(); i++)
+            {
+                std::shared_ptr<beam_rpc_message> currentMessage = m_pendingSend.front();
+                m_pendingSend.pop();
+                send_message(currentMessage);
+            }
+        }
+
+        {
+            static const int numMessageRetries = 10;
+            static const std::chrono::milliseconds messageRetryInterval = std::chrono::milliseconds(10 * 1000); // 10-second retry interval
```

Comment:
we use milliseconds for the timestamps - easier to use milliseconds here than awkward conversion logic below

---
In reply to: [117138035](https://github.com/WatchBeam/interactive-cpp/pull/6#discussion_r117138035) [](ancestors = 117138035)